test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,553 @@
+
+    @cleanup
+    def user_input_weight_not_overwritten(self):
+        self.log.info("Test that the user-input tx weight is not overwritten by the truc maximum")
+
+        tx = CTransaction()
+        tx.version = 3
+
+        self.bulk_tx(tx, 5, int(TRUC_MAX_VSIZE/2))
+
+        assert_raises_rpc_error(
+            -4,
+            "Maximum transaction weight is less than transaction weight without inputs",
+            self.charlie.fundrawtransaction,
+            tx.serialize_with_witness().hex(),
+            {'include_unsafe' : True, 'max_tx_weight' : int(TRUC_MAX_VSIZE/2)}
+        )
+
+    @cleanup
+    def user_input_weight_not_overwritten_v3_child(self):
+        self.log.info("Test that the user-input tx weight is not overwritten by the truc child maximum")
+
+        outputs = {self.alice.getnewaddress() : 10}
+        self.send_tx(self.charlie, [], outputs, 3)
+
+        tx = CTransaction()
+        tx.version = 3
+
+        self.bulk_tx(tx, 5, int(TRUC_CHILD_MAX_VSIZE/2))
+
+        assert_raises_rpc_error(
+            -4,
+            "Maximum transaction weight is less than transaction weight without inputs",
+            self.alice.fundrawtransaction,
+            tx.serialize_with_witness().hex(),
+            {'include_unsafe' : True, 'max_tx_weight' : int(TRUC_CHILD_MAX_VSIZE/2)}
+        )
+
+        self.generate(self.nodes[0], 1)
+        self.alice.fundrawtransaction(tx.serialize_with_witness().hex())
+
+    @cleanup
+    def createpsbt_v3(self):
+        self.log.info("Test setting version to 3 with createpsbt")
+
+        outputs = {self.alice.getnewaddress() : 10}
+        psbt = self.charlie.createpsbt(inputs=[], outputs=outputs, version=3)
+        assert_equal(self.charlie.decodepsbt(psbt)["tx"]["version"], 3)
+
+    @cleanup
+    def send_v3(self):
+        self.log.info("Test setting version to 3 with send")
+
+        outputs = {self.alice.getnewaddress() : 10}
+        tx_hex = self.charlie.send(outputs=outputs, add_to_wallet=False, version=3)["hex"]
+        assert_equal(self.charlie.decoderawtransaction(tx_hex)["version"], 3)
+
+    @cleanup
+    def sendall_v3(self):
+        self.log.info("Test setting version to 3 with sendall")
+
+        tx_hex = self.charlie.sendall(recipients=[self.alice.getnewaddress()], version=3, add_to_wallet=False)["hex"]
+        assert_equal(self.charlie.decoderawtransaction(tx_hex)["version"], 3)
+
+    @cleanup
+    def sendall_with_unconfirmed_v3(self):
+        self.log.info("Test setting version to 3 with sendall + unconfirmed inputs")
+
+        outputs = {}
+        for _ in range(4):
+            outputs[self.alice.getnewaddress()] = 2.00001
+
+        self.send_tx(self.charlie, [], outputs, 2)
+        self.generate(self.nodes[0], 1)
+
+        unspent1 = self.alice.listunspent()[0]
+        unspent2 = self.alice.listunspent()[1]
+        unspent3 = self.alice.listunspent()[2]
+        unspent4 = self.alice.listunspent()[3]
```

Comment:
In f4d209c0e94fad8435215abba3e61d79585d0d12 "test: add truc wallet tests"

Could store the result of `listunspent` and access the specific utxos by index later:

```suggestion
        utxos = self.alice.listunspent()
```

Reply:
Done